grafana/resource_dashboard.go

```diff
@@ -340,5 +344,16 @@ func normalizeDashboardConfigJSON(config interface{}) string {
 	}
 
 	j, _ := json.Marshal(dashboardJSON)
-	return string(j)
+
+	if storeDashboardSHA256 {
+		configHash := sha256.Sum256(j)
+		return fmt.Sprintf("%x", configHash[:])
+	} else {
+		return string(j)
+	}
+}
+
+func MatchStringSHA256(str string) bool {
+	match, _ := regexp.MatchString(`^[A-Fa-f0-9]{64}$`, str)
```

Comment:
By caching at the package level, I think what @inkel meant was to compile the regex once like this:
https://github.com/grafana/terraform-provider-grafana/blob/master/grafana/provider.go#L24-L28

So you could add sha256Regexp to that list and use that pre-compiled regex when matching

Reply:
Yes, that's correct. Sorry I wasn't clear enough! 🙏

Reply:
yes, I actually didn't understand..ok fixed